### PR TITLE
refactor(goal_distance_calculator): rework parameters

### PR DESCRIPTION
**Pull request by PhoebeWu21**
## Description
Implement the ROS Node configuration layout described in https://github.com/orgs/autowarefoundation/discussions/3371 for the goal_distance_calculator package.

- Remove the default value from the source code in order to ensure all parameter values are passed from the parameter files.
- Create the schema

## Tests performed
Package built locally.
`colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release --packages-up-to goal_distance_calculator`

## Effects on system behavior
More reliable and faster parameter configuration file creation.

## Pre-review checklist for the PR author
The PR author **must** check the checkboxes below when creating the PR.
- [x] I've confirmed the [contribution guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/).
- [x] The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).

## In-review checklist for the PR reviewers
The PR reviewers **must** check the checkboxes below before approval.
- [x] The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author
The PR author **must** check the checkboxes below before merging.
- [x] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: common/goal_distance_calculator/launch/goal_distance_calculator.launch.xml [hunk 1]
```diff
@@ -3,6 +3,5 @@
   <arg name="config_file" default="$(find-pkg-share goal_distance_calculator)/config/goal_distance_calculator.param.yaml"/>
   <node pkg="goal_distance_calculator" exec="goal_distance_calculator_node" name="goal_distance_calculator" output="screen">
     <param from="$(var config_file)"/>
-    <param name="oneshot" value="$(var oneshot)"/>
```
Comment (ambroise-arm) [inline common/goal_distance_calculator/launch/goal_distance_calculator.launch.xml:6]: Should `oneshot` at line 2 above also be removed?

Comment (PhoebeWu21) [reply common/goal_distance_calculator/launch/goal_distance_calculator.launch.xml:6]: Done.

**Review by ambroise-arm — APPROVED** [review]

**Review by kaspermeck-arm — APPROVED** [review]
LGTM

File: common/goal_distance_calculator/config/goal_distance_calculator.param.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
 /**:
   ros__parameters:
     update_rate: 10.0
+    oneshot: true
```
Comment (taikitanaka3) [inline common/goal_distance_calculator/config/goal_distance_calculator.param.yaml:4]: @PhoebeWu21  oneshot should not be changed at this PR.

File: common/goal_distance_calculator/src/goal_distance_calculator_node.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2020 Tier IV, Inc. All rights reserved.
+// Copyright 2020-2023 Foxconn, Tier IV, Inc. All rights reserved.
```
Comment (taikitanaka3) [inline common/goal_distance_calculator/src/goal_distance_calculator_node.cpp:1]: @PhoebeWu21 Copy right should not be changed at a small change PR and this PR will not change behavior.

https://github.com/autowarefoundation/autoware.universe/pull/4730#discussion_r1304345840

**Review by taikitanaka3 — CHANGES_REQUESTED** [review]

File: common/goal_distance_calculator/config/goal_distance_calculator.param.yaml (hunk 2 above)
Comment (kaspermeck-arm) [reply common/goal_distance_calculator/config/goal_distance_calculator.param.yaml:4]: @taikitanaka3 
According to the new [parameter coding guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/coding-guidelines/ros-nodes/parameters/), any `declare_parameter` function should have a parameter in the parameter file. Why should the `oneshot` parameter not exists in the parameter file?

File: common/goal_distance_calculator/launch/goal_distance_calculator.launch.xml [hunk 4]
```diff
@@ -1,8 +1,6 @@
 <launch>
-  <arg name="oneshot" default="false"/>
```
Comment (taikitanaka3) [inline common/goal_distance_calculator/launch/goal_distance_calculator.launch.xml:2]: @kaspermeck-arm
I mean one shot parameter is set false by launcher

File: common/goal_distance_calculator/config/goal_distance_calculator.param.yaml (hunk 2 above)
Comment (taikitanaka3) [reply common/goal_distance_calculator/config/goal_distance_calculator.param.yaml:4]: https://github.com/autowarefoundation/autoware.universe/pull/4830/files#r1329414651

or please do not add delete change to launcher files.

File: common/goal_distance_calculator/launch/goal_distance_calculator.launch.xml (hunk 4 above)
Comment (kaspermeck-arm) [reply common/goal_distance_calculator/launch/goal_distance_calculator.launch.xml:2]: @taikitanaka3 
Aha, I see. Thanks for pointing that out!

File: common/goal_distance_calculator/schema/goal_distance_calculator.schema.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      "properties": {
+        "update_rate": {
+          "type": "number",
+          "default": "10.0",
+          "exclusiveMinimum": 0,
+          "description": "Timer callback period. [Hz]"
+        },
+        "oneshot": {
+          "type": "boolean",
+          "default": "true",
```
Comment (kaspermeck-arm) [inline common/goal_distance_calculator/schema/goal_distance_calculator.schema.json:17]: @PhoebeWu21
Set `"default"` to `"false"`.

Comment (kaspermeck-arm) [inline common/goal_distance_calculator/schema/goal_distance_calculator.schema.json:17]: @PhoebeWu21
Set `"default"` to `"false"`.

**Review by kaspermeck-arm — CHANGES_REQUESTED** [review]

File: common/goal_distance_calculator/config/goal_distance_calculator.param.yaml (hunk 2 above)
Comment (PhoebeWu21) [reply common/goal_distance_calculator/config/goal_distance_calculator.param.yaml]: @taikitanaka3 Got it! Done.

File: common/goal_distance_calculator/src/goal_distance_calculator_node.cpp (hunk 3 above)
Comment (PhoebeWu21) [reply common/goal_distance_calculator/src/goal_distance_calculator_node.cpp]: ok!

File: common/goal_distance_calculator/schema/goal_distance_calculator.schema.json (hunk 5 above)
Comment (PhoebeWu21) [reply common/goal_distance_calculator/schema/goal_distance_calculator.schema.json]: Done.

**Review by taikitanaka3 — APPROVED** [review]

**Review by kaspermeck-arm — APPROVED** [review]

**Comment by kaspermeck-arm** [comment]
@mitsudome-r 
Is this PR OK to be merged?

**Pull request merged**